lib/inc/puppetdb-cli/puppetdb-cli.hpp

```diff
@@ -28,10 +28,30 @@ class LIBPUPPETDB_EXPORT PuppetDBConn  {
     PuppetDBConn();
 
     /**
+     * Construct a PuppetDB connection using only config flags
+     * @param urls string of the urls of PuppetDB.
+     * @param cacert string cacert to use for curl.
+     * @param cert client cert to use for curl.
+     * @param key client private key to use for curl.
+     */
+    PuppetDBConn(const std::string& urls,
```

Comment:
Positional args don't make for a great API, we've already got a couple of variants in this patch for example. Have we considered taking a struct instead?